scripts/pre-commit/verify_migration_scripts.py

```diff
@@ -0,0 +1,182 @@
+
+    duplicates = set(new_versions) & set(old_versions)
+    if duplicates:
+        raise MigrationValidationError(
+            f"Duplicate migration versions found: {duplicates}"
+        )
+
+    new_migration_files_set = set(new_migration_files)
+    duplicates = [file for file in new_migration_files_set if new_versions.count(file.split("_")[0]) > 1]
+    if duplicates:
+        raise MigrationValidationError(
+            f"Duplicate migration versions found in new migration files: {duplicates}"
+        )
+
+
+def check_new_migration_file_format(new_migration_files: tp.List) -> None:
+    """
+    Check if the new migration files have a date format in their names.
+    """
+    for file in new_migration_files:
+        if file == BOOTSTRAP_FILE:
+            continue
+        if not re.match(r"^V\d{4}.\d{2}.\d{2}.\d{1,2}__.+\.sql$", file):
+            raise MigrationValidationError(
+                f"New migration file {file} does not have a valid file name format."
+            )
+
+
+def validate_new_migration_dates(new_migration_files: tp.List, old_migration_files: tp.List) -> None:
+    """
+    Validate that the new migration files have a date format in their names.
+    """
+    new_migration_date = [parse_date(file) for file in new_migration_files if file != BOOTSTRAP_FILE]
+    latest_migration_date = [parse_date(file) for file in old_migration_files]
+
+    invalid_new_migration_file_date = [
+        file_date for file_date in new_migration_date if file_date < max(latest_migration_date)
+    ]
+    if invalid_new_migration_file_date:
+        for file_date in invalid_new_migration_file_date:
+            raise MigrationValidationError(
+                f"New migration file date: V{file_date.strftime('%Y.%m.%d')} is older than the latest"
+                f" available migration version date: V{max(latest_migration_date).strftime('%Y.%m.%d')}")
+
+
+def validate_if_bootstrap_file_is_updated(new_migration_files: tp.List) -> None:
+    """
+    Validate if the bootstrap file is updated with the contents of new migration files.
+    """
+    migration_scripts = [f for f in new_migration_files if f != BOOTSTRAP_FILE]
+    if migration_scripts and BOOTSTRAP_FILE not in new_migration_files:
+        raise MigrationValidationError(
+            f"Bootstrap file [{BOOTSTRAP_FILE}] is not updated with the latest migration scripts."
+            " Execute scripts/migrations.sh script to update bootstrap.sql file."
+        )
+
+    if BOOTSTRAP_FILE in new_migration_files:
+        bootstrap_file_path = get_project_root() / METADATA_STORE_SCRIPTS_DIR / BOOTSTRAP_FILE
+        bootstrap_content = Path(bootstrap_file_path).read_text()
+
+        for migration_script in migration_scripts:
+            migration_script_path = get_project_root() / METADATA_STORE_SCRIPTS_DIR / migration_script
+            migration_script_content = Path(migration_script_path).read_text()
+
+            if migration_script_content not in bootstrap_content:
```

Comment:
This is too much. Can we just check for comments that is added by the script tool?